### ISSUE TITLE
Unable to disable queue-thread-local-cleaner-daemon thread 

### ISSUE THREAD

**Issue by cmunger** [OpenHFT/Chronicle-Queue#449]
Hello,

I m using chronicle with a disabled SingleChronicleQueue.SHOULD_RELEASE_RESOURCES flag (chronicle.queue.release.weakRef.resources=false) and since this flag is disabled my understanding is that (let me know if i m wrong) i can also get ride of the thread you create in StoreComponentReferenceHandler so i called StoreComponentReferenceHandler.INSTANCE.close() which unfortunately does not kill the thread correctly, could you please give the user a way to destroy this thread when unneeded ? Thanks

**Comment by dpisklov** [comment]
SingleChronicleQueue.SHOULD_RELEASE_RESOURCES flag only removes cleanup for appender/tailer resources, but the thread in question also cleans up wires.
I'm not sure why do you want to disable cleanup? This will eventually lead you to OOM on off heap space...

**Comment by cmunger** [comment]
well i ended up in this situation because i was using version version 4.6.67 which was working fine, when upgrading to latest version 4.6.77, my same codebase constantly printing warning logs like this : 09:36:53.808 [queue-thread-local-cleaner-daemon] WARN  n.o.c.queue.impl.WireStorePool - Store was not registered 20180327-0736.cq4, when looking at the code i saw that SingleChronicleQueue.SHOULD_RELEASE_RESOURCES would not register stuff that would avoid printing this warning log constantly (which worked) and then i thought that the thread running these cleanup is probably not needed. anyway i'll keep it alive. I still don't know however why i constantly get the wirepool logs when SHOULD_RELEASE_RESOURCES set to true. will i also end up in OOM with SHOULD_RELEASE_RESOURCES = false ?

**Comment by dpisklov** [comment]
@cmunger 
I downgraded that warning to debug.
We recently fixed the memory leak which was causing delayed resource release, now we eagerly release mmapped regions. That *might* be the reason that log statement was printed - we might have already released that file in other place, so this is harmless.

> will i also end up in OOM with SHOULD_RELEASE_RESOURCES = false ?
Yes, that's what I meant.
We wouldn't recommend doing that, this flag is there for testing only.
